### PR TITLE
Modal refactor

### PR DESCRIPTION
**Pull request by theyapps**
Refactored modal to increase re-usability.

### CODE REVIEW

File: spa/src/components/modals/services/ModalService.vue [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<script lang="ts"   >
+import Vue from 'vue';
+
+export default new Vue({
+  methods: {
+    open(component, props = {}) {
+      return new Promise((resolve, reject) => {
+        console.log("ModalService open");
```
Comment (theyapps) [inline spa/src/components/modals/services/ModalService.vue:8]: Remove console log

File: spa/src/components/modals/TestModal.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<template>
```
Comment (theyapps) [inline spa/src/components/modals/TestModal.vue:1]: Delete testmodal

File: spa/src/components/modals/Modal.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+
+export default Vue.extend({
+  name: "Modal",
+  components: {},
+  data: () => {
```
Comment (TheMikeSanto) [inline spa/src/components/modals/Modal.vue:41]: Could shorten this up:
`data: () => ({}),`

File: spa/src/components/modals/mixins/ModalMixin.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export default {
```
Comment (TheMikeSanto) [inline spa/src/components/modals/mixins/ModalMixin.ts:1]: This is awesome!

File: spa/src/components/modals/services/ModalService.vue (hunk 1 above)
Comment (TheMikeSanto) [inline spa/src/components/modals/services/ModalService.vue:8]: Get rid of this log statement

**Review by TheMikeSanto — CHANGES_REQUESTED** [review]
This looks like what we need! A few minor changes to address, and then it's ready to merge. Thank you!!

File: spa/src/components/modals/services/ModalService.vue (hunk 1 above)
Comment (theyapps) [reply spa/src/components/modals/services/ModalService.vue]: Done!

File: spa/src/components/modals/Modal.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        transform
+        -translate-x-1/2
+        border-2
+        border-gray-300
+        max-w-full
+        shadow-lg
+        bg-black
+        overflow-y-hidden" style="width:1024px;">
+      <div class="flex
+      flex-row-reverse
```
[the remark this answers is not in the document]
Comment (theyapps) [reply spa/src/components/modals/Modal.vue]: done

File: spa/src/components/modals/Modal.vue (hunk 3 above)
Comment (theyapps) [reply spa/src/components/modals/Modal.vue]: done

**Comment by TheMikeSanto** [comment]
@theyapps I believe this is ready to merge, but I haven't had time to pull the PR and actually boot it up and test. I will get to that this week, I promise. :)

**Comment by theyapps** [comment]
> @theyapps I believe this is ready to merge, but I haven't had time to pull the PR and actually boot it up and test. I will get to that this week, I promise. :)

Sounds good.

**Comment by dburleson** [comment]
@theyapps has discussed, the code functions well. But, there are a number of coding standards issues, mainly incorrect linebreak style and some semicolons missing.

Once these are resolved, I'll happily get this merged.

**Comment by theyapps** [comment]
> @theyapps has discussed, the code functions well. But, there are a number of coding standards issues, mainly incorrect linebreak style and some semicolons missing.
> 
> Once these are resolved, I'll happily get this merged.

I have updated the line endings and added semicolons

**Pull request merged**